src/handlers/publish.rs

```diff
@@ -102,6 +147,32 @@ pub async fn handle_publish(
         readme: upload.readme.clone(),
         license: Some(pkg_manifest.project.license.clone()),
     };
+
+    let package_name = publish_info.package_name.clone();
+    let package_version = publish_info.num.clone();
+    let source_cid = upload.source_code_ipfs_hash;
+    let abi_cid = upload.abi_ipfs_hash;
+    let dependencies = package_deps
+        .iter()
+        .cloned()
+        .map(PackageDependencyIdentifier::from)
+        .collect();
+    let yanked = false;
+
+    let package_entry = PackageEntry::new(
+        package_name,
+        package_version,
+        source_cid,
+        abi_cid,
+        dependencies,
+        yanked,
+    );
+
+    // Wait for index file insertion to finalize, if it fails we should not
+    // insert the publish information into db.
+    publish_index_file(package_entry).await?;
```

Comment:
I think we should do this after we finish writing to the database so we don't have incomplete data in the DB if this fails.

Reply:
Yeah the other way around is also possible though, if push to index repo fails DB has dangling data. Not sure which one is more manageable, and which one is more fallible.

Reply:
Agreed on the network connection part - i.e. it needs to be verified first before persisting to DB.

Another approach is doing both - i.e. persisting to db with a flag for the state (i.e. publishing); then on successful publish, updating the db record.
Also maybe worthwhile noting what happens if the server abruptly exits during the publish/network operation.
Writing code for this scenario maybe overkill though (unless there is a quick-fix).

Reply:
I think this PR is good way to addresses the issue: https://github.com/FuelLabs/forc.pub/pull/50 - i.e. wrapping the upload in a transaction function.

Reply:
I'll look into #50 first thing in the morning, thanks for that 🙌